docs: update the templates page (#2060)
SwissDataScienceCenter/renku: docs/user/templates.rst

  name>``) (only set when creating a project online in renkulab).

Use custom repositories
^^^^^^^^^^^^^^^^^^^^^^^

If you installed the renku command line interface locally, you can provide
your template repository to the ``renku init`` command. We recommend you
**always** specify a tag (or a commit) when creating a new project from a
custom repository. You can find further details in
`renku init docs <https://renku-python.readthedocs.io/en/latest/commands.html#use-a-different-template>`_.

> If you are using a RenkuLab instance, you can change the `Template source`
> to ``Custom`` on the project creation page. You will be able to insert
> a URL pointing to your template repository.

Reviewer: ```suggestion
If you are using a RenkuLab instance, you can use a custom template repository by
changing the `Template source` to ``Custom`` on the project creation page. There, 
you will be able to insert a URL pointing to your template repository.
```

:ref:`double-check the manifest file<manifest-yaml>`), or the URL/reference
combination is wrong. The UI should show a meaningful error description.

.. note::

  Remember to provide the URL to the **git repository**. For GitHub and
  other code management systems, you can provide the link used to clone
  through ``https``, ending with ``.git``. You can usually leave the
  ``.git`` extension out, but pay particular attention when you try to
  copy-paste directly from your browser. Even an additional final slash can
  lead to an invalid URL, and the error may be confusing.
  This is what you get if you use
  `https://github.com/SwissDataScienceCenter/renku-project-template/` instead
  of `https://github.com/SwissDataScienceCenter/renku-project-template`:

  .. image:: ../_static/images/templates_url_error.png
    :width: 100%
    :align: center
    :alt: Error fetching custom templates

If you think your template may be useful for the broader community, you can
have more visibility by including it in the
`community-contributed project templates repository <https://github.com/SwissDataScienceCenter/contributed-project-templates>`_.

> Feel free to push a pull request and we will validate it.

Reviewer: ```suggestion
Feel free to open a pull request and we will validate it.
```

If you are working in a dedicated RenkuLab deployment and your local
community needs the templates, you should contact the administrators to
include your repository in the RenkuLab template source through the
`renku-values file <https://renku.readthedocs.io/en/latest/admin/index.html#create-a-renku-values-yaml-file>`_.

> Create shareable links with pre-filled fields

Reviewer: ```suggestion
Create shareable project-creation links with pre-filled fields
```

^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> Most of the time, you want your repository to be immediately available
> to the target audience, maybe even having some variables set to specific
> values. We know manually entering the same values, again and again, is a
> tedious process, so we come up with a solution to automatically pre-fill
> some of the fields in the project creation page.
> 
> You can create a link containing metadata embedded in a query parameter.
> Once you enter that link in a browser, the UI will automatically fetch
> the required data and pre-fill the fields for you.

Reviewer: ```suggestion
There are times where you may have many users that each need to create their own
repository, but some of the parameters are shared across all the repositories. An example
could be in a course where each student needs their own repository, but all should use the
same template and metadata.

For this case, it is possible to generate a project-creation link containing 
metadata embedded in a query parameter. When that link is entered in a 
web browser, the UI will  automatically fetch the required information and 
pre-fill the fields.
```

To create a shareable link, start by filling in all the fields as you would
do when creating a new project. Instead of clicking on `Create project`,
click on the dropdown on the right side of the same button and then on
`Create link`.

> You should see a modal where you can select which fields to include in the
> metadata. Some may not be clickable (no value provided), and others are
> deselected by default. The URL updates in real-time, and you can copy it
> for future reference or share it with others.

Reviewer: ```suggestion
You should see a modal dialog where you can select which fields to include in the
metadata. Some may not be clickable (no value provided), and others are
deselected by default. The URL updates in real-time and can be copied to
share with others.
```

.. image:: ../_static/images/templates_shareable_link.png
  :width: 100%
  :align: center
  :alt: Custom template source

.. note::

  You can include any of the listed fields in the link, but you should be
  careful when including namespace and visibility. Your user namespace cannot
  be available to any other users, and group namespaces may require specific
  permission. The visibility is generally tied to the namespace visibility,
  although ``private`` should always be available since it is the most
  restrictive one.
  
  It would be best to prefer fixed references for custom template
  repositories, especially when selecting a template and providing values for
  variables. This means commits and tags are a good choice, while branches
  are not. Otherwise, the template or the variables may change in a later
  version, resulting in a corrupted link.

> Programmatically compute shareable links

Reviewer: This documentation is great and easy to understand, but I wonder if our user documentation is the best place to have it. Would it maybe be better to put it on Discourse and just say:

```
If you need to generate project-creation links yourself, you can see an example of how to do this in [Discourse](link to post).
```

Author: Maybe that part is a bit too technical, but I would still include it here since it was explicitly requested by one of the first users of this feature.
